src/sdk/pynni/nni/metis_tuner/metis_tuner.py

```diff
@@ -0,0 +1,434 @@
+# Copyright (c) Microsoft Corporation
+# All rights reserved.
+#
+# MIT License
+#
+# Permission is hereby granted, free of charge,
+# to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction,
+# including without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and
+# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import copy
+import logging
+import os
+import random
+import statistics
+import sys
+
+from enum import Enum, unique
+from multiprocessing.dummy import Pool as ThreadPool
+
+from nni.tuner import Tuner
+
+import nni.metis_tuner.lib_data as lib_data
+import nni.metis_tuner.lib_constraint_summation as lib_constraint_summation
+import nni.metis_tuner.Regression_GP.CreateModel as gp_create_model
+import nni.metis_tuner.Regression_GP.Selection as gp_selection
+import nni.metis_tuner.Regression_GP.Prediction as gp_prediction
+import nni.metis_tuner.Regression_GP.OutlierDetection as gp_outlier_detection
+import nni.metis_tuner.Regression_GMM.CreateModel as gmm_create_model
+import nni.metis_tuner.Regression_GMM.Selection as gmm_selection
+
+logger = logging.getLogger("Metis_Tuner_AutoML")
+
+@unique
+class OptimizeMode(Enum):
+    '''
+    Optimize Mode class
+    '''
+    Minimize = 'minimize'
+    Maximize = 'maximize'
+
+
+NONE_TYPE = ''
+CONSTRAINT_LOWERBOUND = None
+CONSTRAINT_UPPERBOUND = None
+CONSTRAINT_PARAMS_IDX = []
+
+
+class MetisTuner(Tuner):
+    '''
+    Metis Tuner
+    '''
+
+    def __init__(self, optimize_mode, no_resampling=True, no_candidates=True,
```

Comment:
Since optimize_mode is an "Optional" choice, it's better to set a default value of optimize_mode to "maximize" ?
```suggestion
     def __init__(self, optimize_mode="maximize", no_resampling=True, no_candidates=True,
```